pkg/vault/client.go

```diff
@@ -85,16 +97,99 @@ func (b *BankVaultClient) SetToken(secretPath, token string, log logr.Logger) er
 		return err
 	}
 
-	secretData := secret.Data["data"].(map[string]interface{})
+	secretData, ok := secret.Data["data"].(map[string]interface{})
 
-	if secretData[tokenName] != token {
+	if !ok {
+		secretData = make(map[string]interface{})
+	}
+
+	if !ok || secretData[tokenName] != token {
 
-		log.WithName("vault").Info("secrets don't match, re-applying")
+		b.log.WithName("vault").Info("secrets don't match, re-applying")
 
 		secretData[tokenName] = token
 
+		secret.Data["data"] = secretData
+
 		_, err = b.client.RawClient().Logical().Write(queryPath, secret.Data)
 	}
 
 	return err
 }
+
+// SetDeletePath is needed for Remove to work properly. Set the secretPath before
+// calling Remove().
+func (b *BankVaultClient) SetDeletePath(secretPath string) {
```

Comment:
This effectively limits this client to only write/delete one secret in vault. If in the future we want to manage more thsi becomes a problem.